src/Service/MagicLinks/MagicLinkService.cs

```diff
@@ -1,27 +1,98 @@
 using Passwordless.Common.Services.Mail;
+using Passwordless.Service.Helpers;
 using Passwordless.Service.MagicLinks.Models;
 using Passwordless.Service.Models;
+using Passwordless.Service.Storage.Ef;
 
 namespace Passwordless.Service.MagicLinks;
 
-public class MagicLinkService
+public class MagicLinkService(
+    TimeProvider timeProvider,
+    ITenantStorage tenantStorage,
+    IFido2Service fido2Service,
+    IMailProvider mailProvider)
 {
-    private readonly IFido2Service _fido2Service;
-    private readonly IMailProvider _mailProvider;
-
-    public MagicLinkService(IFido2Service fido2Service, IMailProvider mailProvider)
+    private async Task EnforceLimitsAsync(MagicLinkDTO dto)
     {
-        _fido2Service = fido2Service;
-        _mailProvider = mailProvider;
+        var now = timeProvider.GetUtcNow();
+        var account = await tenantStorage.GetAccountInformation();
+        var accountAge = (now - account.CreatedAt).Duration();
+        var isFreeAccount = account.SubscriptionTier == "Free";
```

Comment:
@Tyrrrz & @abergs The 'SubscriptionTier' in the `Api` is no longer being updated. The `Api` has no concept of what plan a customer is on. The architectural proposal for self-hosting licensing was meant to change all this.

- [PR](https://github.com/bitwarden/passwordless-server/pull/320)

Reply:
Upon longer inspection, you'll need to insert the column `MaxFreeMonthlyLimit` in the `AppFeatures` table.
Then update the appsettings.json to set the values for each plan accordingly.
Then make sure the mapping everywhere happens correctly when an app is created or on a plan change.

Use a good prefix so we know it's used for magic links.

Reply:
Done